cassdegrees/ui/views/student.py

```diff
@@ -100,6 +101,23 @@ def student_create(request):
         response = redirect(student_edit)
         response['Location'] += '?plan='
         return response
+    # If a plan name is given, copy that plan
+    elif plan_to_duplicate:
+        plan = decompress(request.session.get('plan:' + plan_to_duplicate, None))
+        print(plan)
+        if plan:
+            new_plan_name = plan['name'] + ' (Copy)'
+            plan['name'] = new_plan_name
+            plan['date'] = timezone.localtime().strftime('%d/%m/%Y %H:%M')
+            request.session['plan:' + new_plan_name] = compress(plan)
+
+            # Redirect to the student edit page and add the '?plan=' url parameter
+            response = redirect(student_edit)
+            response['Location'] += '?plan=' + new_plan_name
```

Comment:
Can you do this in one call in redirect (e.g. `redirect('?plan=' + new_plan_name)`? Re configuring HTTP headers could potentially be error-prone.

Reply:
Yes I can, but it would need to be `redirect('/edit/?plan='+new_plan_name)`. I figured doing it this way requires fewer rewrites if we want to change the address later, although I might just do it the way you are suggesting as that's used in other areas of the code.